File: src/Notification.fs
module Notification

#if LINUX
open Notifications
#else
open System.Drawing
open System.Windows.Forms
#endif

type Icon =
    | Info
    | Error

let private create icon title body =
    #if LINUX
        let n = new Notification()

        n.IconName <-
            match icon with
            | Info  -> "emblem-sound"
            | Error -> "dialog-error"

        n.Summary <- title
        n.Body <- body
        n.Show() |> ignore
    #else
        use n = new NotifyIcon()

        let sysIcon, toolIcon =
            match icon with
            | Info  -> SystemIcons.Information, ToolTipIcon.Info
            | Error -> SystemIcons.Error, ToolTipIcon.Error

        n.Icon <- sysIcon
        n.Visible <- true
        n.ShowBalloonTip(5000, title, body, toolIcon) |> ignore
    #endif

let createUpdate curIdx maxIdx =
    create
        Info
        (sprintf "Broadcastify Listener Update (%d of %d)" curIdx maxIdx)

let createError = create Error "Broadcastify Update Error"

File: src/Notification.hs
module Notification
    ( create
    , createError
    , createFeedUpdate
    ) where

import Libnotify

type Title = String
type Body = String
type Icon = String

create :: Icon -> Title -> Body -> IO Notification
create icon' title body' =
    display $
        summary title
     <> body body'
     <> icon icon'
     <> timeout Default

createError :: Body -> IO Notification
createError = create "dialog-error" "Broadcastify Update Error"

createFeedUpdate :: Title -> Body -> IO Notification
createFeedUpdate = create "emblem-sound"
